src/common.rs

```diff
@@ -355,6 +355,70 @@ pub struct LoadAvg {
     pub fifteen: f64,
 }
 
+#[cfg(not(target_os = "windows"))]
+pub(crate) mod uid {
+    #[derive(PartialEq, Eq, PartialOrd, Ord, Debug)]
+    pub(crate) struct Uid(libc::uid_t);
+
+    impl Uid {
+        pub(crate) fn new(val: libc::uid_t) -> Self {
+            Self(val)
+        }
+
+        pub(crate) fn as_u32(&self) -> u32 {
```

Comment:
I don't think there is a need for that. Might be more interesting to implement the `Deref` trait instead so you can access the wrapped type from outside.